### PR TITLE
feat(compartment-mapper): Bundler support for aliases

### PR DESCRIPTION
**Pull request by kriskowal**
The Compartment Map contains a bundler that is entirely sufficient for bundling SES proper. Since its original design, @naugtur expanded it to handle CommonJS. So, the missing feature that would make it suitable for bundling a larger bootstrapping system, say, one that could bootstrap eventual-send and import-bundle, was module aliases.

This change introduces an alias map, which fast-forwards every import resolution thru aliases to the concrete implementation of the imported module.

### CODE REVIEW

**Comment by kriskowal** [comment]
cc @erights because this greatly expands the reach of our bundler. We should be able to bundle a bootstrapping layer that uses endo/init, endo/eventual-send, endo/captp, and endo/import-bundle.

cc @kumavis @naugtur because this is probably also relevant for the secure bundler you’re working on.

**Review by michaelfig — APPROVED** [review]
LGTM!  Straightforward.

File: packages/compartment-mapper/src/bundle.js [hunk 1]
```diff
@@ -239,10 +244,21 @@ export const makeBundle = async (read, moduleLocation, options) => {
...
   for (const module of modules) {
     module.indexedImports = Object.fromEntries(
-      Object.entries(module.resolvedImports).map(([importSpecifier, key]) => [
-        importSpecifier,
-        modulesByKey[key].index,
-      ]),
+      Object.entries(module.resolvedImports).map(([importSpecifier, key]) => {
+        key = aliases.get(key) ?? key;
+        const module = modulesByKey[key];
+        if (module === undefined) {
```
Comment (naugtur) [inline packages/compartment-mapper/src/bundle.js:250]: So the original implementation would fail to read the index if an alias was used?

**Review by naugtur — APPROVED** [review]

File: packages/compartment-mapper/src/bundle.js (hunk 1 above)
Comment (kriskowal) [reply packages/compartment-mapper/src/bundle.js:250]: Indeed, the modules array lacks entries for the aliases.

**Pull request merged**